### PR TITLE
[dhcp_server] Fix the issue with "kea-dhcp4.conf" file generation for Smart Switch.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
#### Why I did it
The configuration generated from the template for the Smart Switch contained incorrect data in the "subnet4:id" field. For regular cases, the subnet ID is deduced from the VLAN name. For the Smart Switch, there is always one subnet, and the ID is set to 0.

```
"subnet4": [
        {
            "id": bridge-midplane,
            "subnet": "169.254.200.0/24",
            "pools": [
                {
                    "pool": "169.254.200.1 - 169.254.200.1",
                    "client-class": "sonic:dpu0"
                },
...
```

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Fix configuration generation for Smart Switch device.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

File: src/sonic-dhcp-utilities/dhcp_utilities/dhcpservd/dhcp_cfggen.py [hunk 1]
```diff
@@ -223,8 +223,9 @@ def _construct_obj_for_template(self, dhcp_server_ipv4, port_ips, hostname, cust
                                 "condition": "substring(relay4[1].hex, -{}, {}) == '{}'".format(class_len, class_len,
                                                                                                 client_class)
                             })
+
                     subnet_obj = {
-                        "id": dhcp_interface_name.replace("Vlan", ""),
+                        "id": 0 if smart_switch else dhcp_interface_name.replace("Vlan", ""),
```
Comment (vivekrnv) [inline src/sonic-dhcp-utilities/dhcp_utilities/dhcpservd/dhcp_cfggen.py:228]: Can you provide a description on why this change is needed?

Comment (oleksandrivantsiv) [reply src/sonic-dhcp-utilities/dhcp_utilities/dhcpservd/dhcp_cfggen.py:228]: updated PR description

**Review by vivekrnv — APPROVED** [review]
